### PR TITLE
Migrate model-registry `RestStore` to pytest

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## Related Issues/PRs

 #7201 

## What changes are proposed in this pull request?

Migrates model-registry `RestStore` to pytest.

## How is this patch tested?

- [ ] Existing unit/integration tests
- [x] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/store/model_registry/test_rest_store.py [hunk 1]
```diff
@@ -33,327 +32,353 @@
...
-        request_mock.return_value = response
-        yield request_mock
-
-
-def mock_http_request(f):
-    @functools.wraps(f)
-    @mock.patch(
+@contextmanager
+def mock_http_request_200():
+    with mock.patch(
```
Comment (WeichenXu123) [inline tests/store/model_registry/test_rest_store.py:37]: We can move this to utils module and all tests can use it.

File: tests/store/model_registry/test_rest_store.py [hunk 2]
```diff
@@ -33,327 +32,353 @@
...
+        yield m
 
-    return wrapper
 
-
-def mock_multiple_http_requests(f):
-    @functools.wraps(f)
-    @mock.patch(
+@contextmanager
+def mock_http_request_403_200():
```
Comment (WeichenXu123) [inline tests/store/model_registry/test_rest_store.py:45]: ditto

File: tests/store/model_registry/test_rest_store.py [hunk 3]
```diff
@@ -33,327 +32,353 @@
...
+        {"filter_string": "model = 'yo'"},
+        {"max_results": 400},
+        {"page_token": "blah"},
+        {"order_by": ["x", "Y"]},
+    ]
+    # Generate all combinations of the above params
+    for sz in [0, 1, 2, 3, 4]:
+        for combination in combinations(params_list, sz):
+            params = {k: v for d in combination for k, v in d.items()}
+            yield params
```
Comment (WeichenXu123) [inline tests/store/model_registry/test_rest_store.py:160]: Can we use `@pytest.mark.parametrize` instead to generate the combinations ?

Comment (harupy) [reply tests/store/model_registry/test_rest_store.py:160]: This function is not a test. See line 163.

File: tests/store/model_registry/test_rest_store.py (hunk 2 above)
Comment (harupy) [reply tests/store/model_registry/test_rest_store.py]: Fixed

File: tests/store/model_registry/test_rest_store.py (hunk 1 above)
Comment (harupy) [reply tests/store/model_registry/test_rest_store.py]: Fixed

File: tests/store/model_registry/test_rest_store.py (hunk 3 above)
Comment (harupy) [reply tests/store/model_registry/test_rest_store.py:139]: I think we can.

Comment (WeichenXu123) [reply tests/store/model_registry/test_rest_store.py]: I think we can, each param we can set 2 options [some_value, None], None means we don't set the param when calling related api

Comment (harupy) [reply tests/store/model_registry/test_rest_store.py]: See https://github.com/mlflow/mlflow/pull/7780/commits/a7f8a024a1cf931c36de4a0bf363cba6dc7158f0

File: tests/store/model_registry/test_rest_store.py [hunk 4]
```diff
@@ -125,22 +123,19 @@ def test_search_registered_models(store, creds):
...
+@pytest.mark.parametrize("page_token", [None, "blah"])
+@pytest.mark.parametrize("order_by", [None, ["x", "Y"]])
+def test_search_registered_models_params(
+    store, creds, filter_string, max_results, page_token, order_by
+):
+    params = {
+        **({"filter_string": filter_string} if filter_string else {}),
+        **({"max_results": max_results} if max_results else {}),
+        **({"page_token": page_token} if page_token else {}),
+        **({"order_by": order_by} if order_by else {}),
```
Comment (WeichenXu123) [inline tests/store/model_registry/test_rest_store.py:137]: nit:
```
params = {
   "filter_string": filter_string,
   "max_results": max_results
   ...
}

params = {k: v for k, v in params.items() if v}
```

**Review by WeichenXu123 — APPROVED** [review]

File: tests/store/model_registry/test_rest_store.py (hunk 4 above)
Comment (harupy) [reply tests/store/model_registry/test_rest_store.py]: Nice approach!

**Pull request merged**